src/main/java/com/gmail/inverseconduit/bot/InteractionBot.java

```diff
@@ -26,13 +26,12 @@ public InteractionBot(ChatInterface chatInterface) {
 
     @Override
     public void start() {
-        executor.scheduleAtFixedRate(this::processInteractions, 200, 700, TimeUnit.MILLISECONDS);
+        // asynchronously enqueue the processing by blocking supplier
+        processingThread.submit(this::processInteractions);
```

Comment:
Wait wait wait; I don't know the intent behind the code there; but if indeed a timer is used for polling then a `BlockingQueue<?>` would indeed be more appropriate here, and again, that's imho.